src/NPCNameProvider.cpp

```diff
@@ -0,0 +1,29 @@
+#include "NPCNameProvider.h"
+#include "NND_API.h"
+
+const char* NPCNameProvider::GetName(const RE::Actor* actor) const
+{
+	if (NND) {
+		if (auto name = NND->GetName(actor, NND_API::NameContext::kSubtitles); !name.empty()) {
+			return name.data();
+		}
+	}
+
+	if (auto xTextData = actor->extraList.GetByType<RE::ExtraTextDisplayData>(); xTextData) {
```

Comment:
Also, I was wondering is there a reason you get the name from `extraList` manually, rather than calling `actor->GetDisplayFullName()` or a better yet `actor->GetActorBase()->GetShortName()` which original SubtitleManager used?

Reply:
Yes, it looks like I should use `GetDisplayFullName()`; I didn't know about it and I even asked about this in the discord. I'm not sure what you mean about `GetShortName()`, I can't find that function in clib and I don't know what to look for in the executable. From what I can tell, the vanilla game uses `GetDisplayFullName()` in its own handling of subtitles, but maybe a different branch leads to `GetShortName`

Reply:
Yeah, the game does this:
![image](https://github.com/WaterFace/subtitles/assets/3248600/03f200c1-15ff-4e56-ab1c-98f40fa72e43)

So what it does is getting the Short Name record of NPC:
![image](https://github.com/WaterFace/subtitles/assets/3248600/3325f07a-1799-4e05-ac90-23822ff612df)
which fits more tightly into subtitles.

So if there is no particular reason work with `extraList` manually, I'd recreate the same logic that original subtitle manager did with short names.